### PR TITLE
Spelling mis-correction transform

### PR DESCRIPTION
**Pull request by nick11roberts**
This transformation is a mechanism for data augmentation which applies (neural) spelling correction to sentences whose words have been so heavily misspelled that spelling correction fails to recover the original word. This is inspired by text messages I often receive from certain anonymous sources. 

@vinayprabhu @rteehas @jjangsangy

### CODE REVIEW

**Comment by Mayukhga83** [comment]
Hi @nick11roberts,
I was assigned to review your transformation. Thank you for the contribution. I would brief my findings

Correctness: All checks failed
Interface: I think correct.

Description: The README is not clear about the goal of the transformation. Firstly it says that it applies (neural) spelling correction to sentences whose words have been so heavily misspelled, but actually it does the opposite. The test cases shows that it takes a sentence and changes the structure in a nonsensical manner

Applicable Tasks & Keywords: Relevant
Specificity: transforms sentence
Novelty: ok
Adding New Libraries: No
Data and code source: Data and code well documented
Test Cases: 10 have been added (I think its good)
Evaluating Robustness: Not performed yet
Languages other than English: No

Note> The README should be edited. Moreover I think the transformation has nothing to do with mis-spellings, it just modifies the structure in a nonsensical way. Therefore the nomenclature is not appropriate

**Comment by nick11roberts** [comment]
Thank you for the feedback, and I think this indicates that the readme as it stands failed to properly communicate the contribution of this task. The idea of this transformation is to force the spelling correction algorithm to make mistakes - in other words, misspellings are artificially made to be so bad that the spelling correction algorithm chooses the **wrong** correction. Hence the nomenclature - "spelling _mis_-correction" as opposed to "spelling correction." Spelling correction is used as a subroutine here - the goal of the task is not to perform spelling correction on misspelled sentences. 

Said another way, the source sentence is counterfactually modified to contain heavy enough misspellings to cause errors in spelling correction. 

Indeed, the nonsensical output of the transformation is _intentional_. These types of spelling-correction errors occur in real life, and would therefore reasonably occur in real-world datasets. Furthermore, the structure of the sentence is not modified. The modification occurs on a per-word basis. 

I would be happy to add clarifying examples of the process by which these spelling correction mistakes occur in the readme.

File: transformations/spelling_mis_correction/transformation.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from neuspell import BertChecker 
+# TODO remember to add neuspell to requirements.txt
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+def modified_butter_finger(text, prob=0.1, keyboard="querty", seed=0,
```
Comment (samsontmr) [inline transformations/spelling_mis_correction/transformation.py:14]: Could you mention what was modified?

**Review by samsontmr — COMMENTED** [review]
This is a very nice idea! The build appears to be failing though.

Additionally, could you refactor the functions so that `spelling_mis_correction` operates on words instead of the entire sentence? In other words, he loop over the words in the sentence should be in `generate`.

**Comment by AbinayaM02** [comment]
Hi @nick11roberts : To fix the build failure, please pull the latest code and add your transformation name in the test/mapper.py file in the appropriate dictionary. Since yours is a heavy transformation, please set the heavy flag to True in your transforamtion.py file and run the pytest separately and confirm if it passes [The workflow checks the test cases for only light transformations/filters].

**Comment by kaustubhdhole** [comment]
Feel free to reopen after the issues have been fixed.

**Pull request closed without merging**